src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java

```diff
@@ -180,11 +184,35 @@ public static void validateFieldSchemaMatches(Field bqField, Schema.Field field,
       return;
     }
 
-    if (TYPE_MAP.get(type) == null) {
+    if (TYPE_MAP.get(type) == null && !SUPPORTED_COMPLEX_TYPES.contains(type)) {
```

Comment:
seems simpler to have a SUPPORTED_TYPES set that contains all the supported types, rather than having multiple places that check for simple type or supported complex type. Unless they are distinguished for some reason?

Reply:
We have a separate set of supported complex types because there is no corresponding `LegacySQLTypeName` type mapping for array type.